src/v/cluster/partition_balancer_planner.cc

```diff
@@ -163,6 +163,61 @@ class partition_balancer_planner::request_context {
     ss::abort_source& _as;
 };
 
+/// Following two methods leak some of the space mgmt abstractions
+/// into the balancer code. That is something that needs fixing.
+/// The state is complicated because of upgrade considerations
+/// and whether the node space management has all the required
+/// information to send in a help report. These combinations
+/// are captured in a tristate + feature flag to account for
+/// all scenarios.
+
+uint64_t partition_balancer_planner::get_node_total_size_bytes(
+  const node::local_state& node_state) {
+    if (unlikely(!_state.is_trimming_space_management_active())) {
```

Comment:
This looks pretty complicated. I wonder if we can ignore what the feature flag state is saying on our side and just trust what the node itself sent us?

Reply:
Redid this part after discussing with Noah.